pkg/kv/local/iterator.go

```diff
@@ -39,6 +42,10 @@ func (e *EntriesIterator) Next() bool {
 	if e.err != nil {
 		return false
 	}
+	if e.ctx.Err() != nil {
```

Comment:
not recommended to check the ctx Err. In this specific case there is a switch case that we can add ctx done check and break/err as needed.

Reply:
It's a switch, not a select. I can move the `e.ctx.Err() != nil` check there but not `<-e.ctx.Done()`